readme/README.md

```diff
@@ -241,6 +242,25 @@ If you want banned import analysis but without breaking your build you can set
     </rules>
 </configuration>
 ```
+## Exclude source roots
+By default, all source roots reported by Maven is subject to the banned import checks, which for example includes but
+is not limited to `${project.basedir}/src/main/java`, `${project.basedir}/src/test/java`,
+`${project.basedir}/target/generated-sources/main/java` and
+`${project.basedir}/target/generated-test-sources/main/java`. You can exclude source roots using the
+`excludedSourceRoot(s)` option. Note that absolute path is required.
+```xml
+<configuration>
+    <rules>
+        <restrictImports implementation="de.skuzzle.enforcer.restrictimports.rule.RestrictImports">
+            <excludedSourceRoots>
+                <excludedSourceRoot>${project.basedir}/target/generated-sources/main/java</excludedSourceRoot>
```

Comment:
The `${project.basedir}` expression must be escaped, otherwise it leads to user specific absolute paths in the `README.md` that is generated during the release process (see the changed files above).
Not sure how right now what is the proper escape character there. I think either `\$` or `$$`?

Also, can't you just leave out the `${project.basedir}` part in the configuration? I think that would be desirable from a user's perspective.

Reply:
Do you mean we can also support relative path and try to expand it? Yeah we can do it. I also realized `${project.build.directory}` is a better choice than `${project.basedir}/target`.

Reply:
I made a change so it support both absolute and relative paths.